packages/@react-aria/overlays/src/useOverlayTrigger.ts

```diff
@@ -45,6 +46,26 @@ export function useOverlayTrigger(props: OverlayTriggerProps, state: OverlayTrig
     }
   });
 
+  useEffect(() => {
+    if (isOpen === true && visibleOverlays.length > 1) {
```

Comment:
@devongovett yep, I think I had it working for one, then changed for the other and forgot to retest. I can reproduce it and I can fix it for one or the other. I'm having trouble solving for both though. It's this line here
having it as 1 solves it for the hooks which have no delay
having it as 0 solves it for the components which have the OpenTransition with a delay of 0 (which gets queued)

without knowing the overlay this trigger is opening, it's hard to determine if it's in the visibleOverlays or not yet

Some thoughts on it so far:
I could pass in overlayRef, if it's undefined then i know we're dealing with the delayed case, but that changes the API unless I make it optional and then because it's checking undefined it's hard to know if it's omitted or just that it's not mounted yet.

We can't check the id of the last visible overlay, the id isn't on the ref, it's somewhere further down in the tree.

We can't pass the id into useOverlay for storage in the array of visibleOverlays because it's on the dialog or menu context, so we'd need to change an API to get it to the useOverlay hook.

I thought about waiting to remove previous overlays dependent on when the new overlay mounts, but if there's a delay bigger than 0 to show it, then that might get a little funny looking? I think it'd be messy to implement as it'd be a new piece of state i think.